src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodDefHandler.java

```diff
@@ -107,11 +99,24 @@ private static int getMethodDefLineStart(DetailAST mainAst) {
 
     @Override
     public void checkIndentation() {
-        checkModifiers();
+        final int firstNodeColumnNo = expandedTabsColumnNo(getMainAst());
```

Comment:
See this, https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/331cd0c_2021112207/reports/diff/hibernate-orm/index.html#A1

 I guess the problem is due to we are using `expandedTabsColumnNo()`, we can used mix of that with `getLineStart()`.

Btw, everthing else is looking fine and your code is awesome too.

Reply:
@Abhishek-kumar09 Thank you :-) I used a config with default tabWidth (= 8) for the last report. I will generate a new one with tabWidth=4.

Reply:
@Abhishek-kumar09 Please check new report